File: Tier1/Shared/DTOs/HousingListDTO.cs
using System.Runtime.CompilerServices;
using Shared.Domain;

namespace Shared.DTOs;

public class HousingListDTO
{
    public List<Housing> HousingList;
    public string ErrorMessage;

    public HousingListDTO(List<Housing> housingList, string errorMessage)
    {
        HousingList = housingList;
        ErrorMessage = errorMessage;
    }
}

File: Tier2/src/main/java/via/sep3/group11/tier2/shared/DTOs/HousingListDTO.java
package via.sep3.group11.tier2.shared.DTOs;

import via.sep3.group11.tier2.shared.domain.Housing;

import java.util.List;

public class HousingListDTO {
    private List<Housing> housingList;
    private String errorMessage;

    public HousingListDTO(List<Housing> housingList, String errorMessage) {
        this.housingList = housingList;
        this.errorMessage = errorMessage;
    }

    public List<Housing> getHousingList() {
        return housingList;
    }

    public String getErrorMessage() {
        return errorMessage;
    }
}
